### DISCUSSION TITLE
Excalibur Router

### DISCUSSION THREAD

**Discussion by mattjennings** [excaliburjs/Excalibur#2423] [Show and tell]
[Excalibur Router](https://github.com/mattjennings/excalibur-router) is a router with support for scene transitions and loading scenes. The README covers it all in more depth so I won't repeat it here, but give it a look if it interests you!

[Here's an example showing off all the bells and whistles.](https://codesandbox.io/s/github/mattjennings/excalibur-router/tree/main/examples/with-multiple-loaders?file=/src/index.ts)

Some background: it's been the underlying router for a meta framework [Merlyn](https://github.com/mattjennings/merlyn) that I've been working on, I just finally got around to pulling out and putting it together as its own nice little package. The API is still a little rough but I think the bulk of it is there. If you have any suggestions I'm all ears!

**Comment by eonarheim** [comment]
@mattjennings this is so cool! Definitely going to be trying this out soon!
